gw_eccentricity/eccDefinition.py

```diff
@@ -981,11 +1054,94 @@ def get_default_extrema_finding_kwargs(self, width):
             "rel_height": 0.5,
             "plateau_size": None}
         return default_extrema_finding_kwargs
+    
+    def get_default_special_interp_kwargs_for_extrema(self):
+        """Get default kwargs to be passed to extrema interpolating method.
+        
+        Accurate interpolation of omega_gw extrema is crucial for obtaining
+        monotonic eccentricity evoultion with time. Depending on the
+        interpolation method, this function returns the default kwargs to be
+        passed to the interpolating function. 
+        """
+        allowed_methods = self.extra_kwargs["special_interp_kwargs_for_extrema"].keys()
+        if self.omega_gw_extrema_interpolation_method == "spline":
+            kwargs = get_default_spline_kwargs()
+        elif self.omega_gw_extrema_interpolation_method == "rational_fit":
+            kwargs = get_default_rational_fit_kwargs()
+        else:
+            raise Exception("Unknown omega_gw extrema interpolation method "
+                            f"`{self.omega_gw_extrema_interpolation_method}`. "
+                            f"Allowed methods are {allowed_methods}")
+        return kwargs
+    
+    def check_special_interp_kwargs_for_extrema(self, extra_kwargs):
+        """Check extrema interp kwargs provided in extra_kwargs.
+        
+        The `special_interp_kwargs_for_extrema` should have exactly one
+        key matching the current `omega_gw_extrema_interpolation_method`.
```

Comment:
If that is the case, why even leave special_interp_kwargs_for_extrema as an option for the user?